leakcanary-android-proguard/src/main/java/com/squareup/leakcanary/proguard/LeakcanaryProguardPlugin.kt

```diff
@@ -0,0 +1,42 @@
+package com.squareup.leakcanary.proguard
+
+import com.android.build.gradle.AppExtension
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+
+class LeakcanaryProguardPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    val extension = project.extensions.create("leakCanary", LeakCanaryProguardExtension::class.java)
+
+    val androidPluginHandler = { _: Plugin<*> ->
+      val android = project.extensions.getByType(AppExtension::class.java)
+
+      project.afterEvaluate {
+        android.applicationVariants.all { variant ->
+
+          if (extension.obfuscatedVariantNames?.contains(variant.name) == true) {
+            val copyProguardMappingFileTask = project.tasks.create(
```

Comment:
Here you want to use Task Configuration Avoidance instead: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html#sec:how_do_i_defer_creation.

Reply:
Didn't know about that, thank you!